cms/admin/utils.py

```diff
@@ -696,6 +696,48 @@ def save_model(self, request: HttpRequest, obj: models.Model, form: forms.Form,
             setattr(form._content_instance, self.grouper_field_name, obj)
             form._content_instance.save()
 
+    def get_search_fields(self, request):
+        """Return search fields for either grouper model or content model"""
+        content_search_fields = []
+        grouper_search_fields = []
+        for field_name in self.search_fields:
+            if field_name.startswith(CONTENT_PREFIX):
+                content_search_fields.append(field_name.replace(CONTENT_PREFIX, ""))
+            else:
+                grouper_search_fields.append(field_name)
+
+        if getattr(request, "_content_fields", False):
+            return content_search_fields
+
+        return grouper_search_fields
+
+    def get_search_results(self, request, queryset, search_term):
+        # Reset _content_fields flag
+        request._content_fields = False
+        grouper_search_result, may_have_duplicate_grouper = super().get_search_results(request, queryset, search_term)
+
+        # Set _content_fields for get_search_fields to return content model search fields
+        request._content_fields = True
+        search_result_from_content, may_have_duplicate_content = self._get_content_search_result(
+            request, queryset, search_term
+        )
+
+        return grouper_search_result | search_result_from_content, (
+            may_have_duplicate_grouper & may_have_duplicate_content
+        )
+
+    def _get_content_search_result(self, request, queryset, search_term):
+        """Get search results from content model"""
+        content_queryset = self.content_model.admin_manager.all()
+        if self.get_search_fields(request):
+            content_search_result, __ = super().get_search_results(request, content_queryset, search_term)
+        else:
+            content_search_result = self.content_model.admin_manager.none()
```

Comment:
```suggestion
        try:
            # Set flag on request object to get the content search fields. `get_search_results` will call 
            # `get_search_fields` to get the content search fields.
            request._content_fields = True
            content_search_result, __ = super().get_search_results(request, content_queryset, search_term)
        finally:
            request._content_fields = False
```
Setting the `_content_fields` flag in a try finally block will ensure that it is reset even if an exception occurs.

Also, the admin's `get_search_results` will evaluate the search fields - no need to go through the list twice.

Reply:
I might be wrong about the second point here: If the search list is empty the `super().get_search_results()` call might return **all** content objects - we only want additional ones.